linkerd/router/src/layer.rs

```diff
@@ -56,29 +54,30 @@ impl Config {
 
 // === impl Layer ===
 
-pub fn layer<Rec, Req>(config: Config, recognize: Rec) -> Layer<Req, Rec>
+impl<Req, Rec> Layer<Req, Rec>
 where
     Rec: Recognize<Req> + Clone + Send + Sync + 'static,
 {
-    Layer {
-        config,
-        recognize,
-        _p: PhantomData,
+    pub fn new(config: Config, recognize: Rec) -> Self {
+        Self {
+            config,
+            recognize,
+            _p: PhantomData,
+        }
```

Comment:
what motivated the change from `router::layer` to `router::Layer::new`? seems like the diff would've been smaller without this?

not a big deal, just curious.

Reply:
Eh, it just gets awkward since it would actually be `router::layer::layer` or it would overload the name or something... As we split things out I think it just generally makes sense to adopt tower best practices.